### DISCUSSION TITLE
Build error with custom installed openssl

### DISCUSSION THREAD

**Discussion by Shwan-Kim** [open-quantum-safe/oqs-provider#604] [Q&A] [answered]
I upgraded from openssl 1.1.1 to openssl 3.2.3, then built oqs-provider with OPENSSL_ROOT_DIR option.
Install openssl ./config --prefix=usr/local/ssl/ --openssldir=usr/local/ssl/
But it is not working with errors. 
How can I fix it?

~/PQC/oqs-provider-main$ openssl version
OpenSSL 3.2.3 3 Sep 2024 (Library: OpenSSL 3.2.3 3 Sep 2024)

~/PQC/oqs-provider-main$ sudo cmake -S . -B _build && cmake --build _build -DOPENSSL_ROOT_DIR="usr/local/ssl/"
CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

-- Creating Debug build with OQS-Debug env vars enabled
-- Build will store public keys in PKCS#8 structures
CMake Error at /snap/cmake/1435/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR: Found unsuitable version "1.1.1", but
  required is at least "3.0" (found /usr/local/ssl/lib64/libcrypto.so, )
Call Stack (most recent call first):
  /snap/cmake/1435/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:601 (_FPHSA_FAILURE_MESSAGE)
  /snap/cmake/1435/share/cmake-3.31/Modules/FindOpenSSL.cmake:691 (find_package_handle_standard_args)
  CMakeLists.txt:51 (find_package)

-- Configuring incomplete, errors occurred!

Also, I try to use fullbuild script with OPENSSL_INSTALL option.
I encountered an error like the one below. 
please let me know how to fix it.

root@sr100-400T8A-400S8A:/home/shkim/PQC/oqs-provider-main/scripts# OPENSSL_INSTALL="/usr/local/ssl/" ./fullbuild.sh
need to re-build static liboqs...
cloning liboqs main...
Cloning into 'liboqs'...
remote: Enumerating objects: 2579, done.
remote: Counting objects: 100% (2579/2579), done.
remote: Compressing objects: 100% (1260/1260), done.
remote: Total 2579 (delta 1570), reused 1763 (delta 1308), pack-reused 0 (from 0)
Receiving objects: 100% (2579/2579), 4.13 MiB | 9.86 MiB/s, done.
Resolving deltas: 100% (1570/1570), done.
-- The C compiler identification is GNU 9.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CC_SUPPORTS_WA_NOEXECSTACK
-- Performing Test CC_SUPPORTS_WA_NOEXECSTACK - Success
-- Performing Test LD_SUPPORTS_WL_Z_NOEXECSTACK
-- Performing Test LD_SUPPORTS_WL_Z_NOEXECSTACK - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Alg enablement unchanged
-- Found OpenSSL: /usr/local/ssl/lib64/libcrypto.so (found suitable version "3.2.3", minimum required is "1.1.1")
-- Looking for aligned_alloc
-- Looking for aligned_alloc - found
-- Looking for posix_memalign
-- Looking for posix_memalign - found
-- Looking for memalign
-- Looking for memalign - found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - found
-- Looking for explicit_memset
-- Looking for explicit_memset - not found
-- Looking for memset_s
-- Looking for memset_s - not found
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") found components: doxygen dot
-- Configuring done (1.3s)
-- Generating done (0.3s)
-- Build files have been written to: /home/shkim/PQC/oqs-provider-main/scripts/liboqs/_build
[1678/1678] Linking C executable tests/dump_alg_info
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/cmake/liboqs/liboqsConfig.cmake
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/cmake/liboqs/liboqsConfigVersion.cmake
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/pkgconfig/liboqs.pc
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/liboqs.a
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/cmake/liboqs/liboqsTargets.cmake
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/lib/cmake/liboqs/liboqsTargets-noconfig.cmake
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/oqs.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/aes_ops.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/common.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/rand.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sha2_ops.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sha3_ops.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sha3x4_ops.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_stfl.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_bike.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_frodokem.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_ntruprime.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_classic_mceliece.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_hqc.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_kyber.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/kem_ml_kem.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_dilithium.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_ml_dsa.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_falcon.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_sphincs.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_mayo.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/sig_cross.h
-- Installing: /home/shkim/PQC/oqs-provider-main/scripts/.local/include/oqs/oqsconfig.h
oqsprovider (_build/lib/oqsprovider.so) not built: Building...
CMake Error: The source directory "/home/shkim/PQC/oqs-provider-main/scripts" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
provider build failed. Exiting.
root@sr100-400T8A-400S8A:/home/shkim/PQC/oqs-provider-main/scripts# cd ..
root@sr100-400T8A-400S8A:/home/shkim/PQC/oqs-provider-main# OPENSSL_INSTALL="/usr/local/ssl/" ./scripts/fullbuild.sh
oqsprovider (_build/lib/oqsprovider.so) not built: Building...
CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

-- Creating Debug build with OQS-Debug env vars enabled
-- Build will store public keys in PKCS#8 structures
CMake Error at /snap/cmake/1435/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR: Found unsuitable version "1.1.1", but
  required is at least "3.0" (found /usr/local/ssl/lib64/libcrypto.so, )
Call Stack (most recent call first):
  /snap/cmake/1435/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:601 (_FPHSA_FAILURE_MESSAGE)
  /snap/cmake/1435/share/cmake-3.31/Modules/FindOpenSSL.cmake:691 (find_package_handle_standard_args)
  CMakeLists.txt:51 (find_package)

-- Configuring incomplete, errors occurred!
provider build failed. Exiting.

**Comment by baentsch** [comment, marked as answer]
This error message points to an incorrect setup:

> system variable OPENSSL_ROOT_DIR: Found unsuitable version "1.1.1", but
required is at least "3.0" (found /usr/local/ssl/lib64/libcrypto.so, )

Please consider using our setup script (scripts/fullbuild.sh) to avoid any such problems.

**Reply by Shwan-Kim** [reply]
Thanks for your reply. I have resolved the problem after reading the scripts!
